### DISCUSSION TITLE
Allow timed featured when using workflows

### DISCUSSION THREAD

**Discussion by alve89** [joomla/joomla-cms#47129] [3. Approved Features]
### Steps to reproduce the issue
When using workflows for articles, an article can't have a timed featured, so you can't set a start or a finish for the article to be featured.

### Expected result
This actually makes sense because you can't toggle the featured status manually but only via the workflows / transitions. But this increases the effort of managing all content by far because the manager have to check all the old articles which are still featured.

So the only idea I had was not just start / finish featured (when using workflows), but set a start/finish for any transition.

### System information (as much as possible)
J4.0.0

# Approval Information
Approved: 18.03.2026
Minimum Target Version: 6.2

**Comment by richard67** [comment]
@alve89 Do I get it right and this issue is a feature request?

There is a Summer of Code 2021 project for integrating a scheduler into Joomla 4.1: #35143 .

As far as I know there are ideas to use that also for scheduled workflow transitions.

Maybe @bembelimen can tell better what the plans are with that.

**Comment by brianteeman** [comment]
No this is nothing to do with the gsoc scheduler.

**Comment by alve89** [comment]
Just to be clear and to avoid any misunderstandings: I mean the problem, that this times aren't stored (and therefore also not used) when using workflows:

![image](https://user-images.githubusercontent.com/11487054/130442914-f9eeef0b-7306-408f-93bc-833222b982b6.png)

**Comment by brianteeman** [comment]
The cause of this is the same as the one reported here.
#29942

**Comment by bembelimen** [comment]
The idea with timed transition execution for the workflow was floating around for a while in our heads. The biggest challenge was: how can we support this feature on small webhosts without cronjob functionality, that was the main reason, why https://github.com/joomla/joomla-cms/pull/35143 was started.

But some more functionallity is needed, like a workflow API ( + Scheduler plugin) and an interface to manage the timing in articles etc. 

So it all depends, if we're finding developer/designer who implement it...

**Comment by brianteeman** [comment]
@bembelimen It doesnt need cronjobs or scheduled tasks. It will work just as simply as the current non-workflow system would work. The problem is that both of them have a bug that you cannot save a date unless you already made it featured

**Comment by bembelimen** [comment]
I know, but what I would like to have is that there are e.g. repeating events and someone creates an article, where he posts some stories for this day.
Now the user can go into the system and set up: article should be published 3 days before the event, 1 day before it should be featured, 1 day after unfeatured and the next day unpublished. This should be automatically repeated every year without any manual interaction.

There you would need the scheduler and you're much more flexible/automated.

**Comment by bembelimen** [comment]
Btw. this feature up/down function is really awful, as bad as the publish up/down, which also only works when the article is published (and does not publish up from unpublish)....

**Comment by alve89** [comment]
@bembelimen I really like this idea of a "scheduler". And I understand the basic problem @brianteeman: If you're using workflows, you can't just make an article scheduled featured because this could need a new stage / transition of the workflow.

But: As long as this isn't implemented (and it doesn't sound like "only a little effort") I agree with @brianteeman - may be you just can enable

1. Saving times for scheduling
2. Triggering them as it works the non-workflow-way

It's not a clean way (actually it's quite qad) but the "real" scheduler will take quite a long time until it is implemented (mainly because it's a new feature, while there still are bugs which is more important to fix).

So: The worst thing that could happen is that an article is automatically featured without an transition / without setting a new stage.

**Comment by bembelimen** [comment]
It's part of this years [GSOC](https://community.joomla.org/gsoc/projects-2026.html) project

**Comment by HLeithner** [comment]
**The Joomla Maintainer Team decided at the 18.03.2025 meeting to accept this feature.**
